### PR TITLE
feat(components): Add description to Checkbox

### PR DESCRIPTION
**Pull request by eddysims**
## Motivations

We have started to use small descriptions under checkboxes. To keep these as a consistent pattern we will add this as a prop to `Checbox`.

## Changes

Adds a `description` prop to the `Checkbox` component

### Added

- `description` prop to the `Checkbox` component

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Review by seanhealy — APPROVED** [review]

**Comment by rebecca-li** [comment]
In the proposal I was thinking all helper text would be slightly smaller (14px) than our regular body text: https://www.figma.com/file/bNQJHfXPgTyjt85jSW0BUG/Design-System-Contribution-Helper-text?node-id=26%3A2

File: packages/components/src/Checkbox/Checkbox.mdx [hunk 1]
```diff
@@ -27,6 +27,7 @@ import { Checkbox } from "@jobber/components/Checkbox";
         checked={checked}
         label="Save card for future use"
         onChange={setChecked}
+        description="If you save your card now, you'll never need to save it again."
```
Comment (chris-at-jobber) [inline packages/components/src/Checkbox/Checkbox.mdx:30]: This doesn't have to block this PR, but could we pull the description from the default example and highlight it later on a la **Indeterminate**? I think the first example of a component that we show should focus on the most likely implementation.

File: packages/components/src/Checkbox/Checkbox.css [hunk 2]
```diff
@@ -66,6 +72,14 @@
...
-  display: inline-block;
   margin: 0 var(--space-small);
 }
+
+.description {
+  margin-top: var(--space-smaller);
+  margin-left: var(--space-large);
+}
+
+.description > p {
```
Comment (eddysims) [inline packages/components/src/Checkbox/Checkbox.css:83]: I have added a comment in an upcoming backlog ticket to replace this with an updated version of `Text`

**Comment by rebecca-li** [comment]
LGTD

**Review by rebecca-li — APPROVED** [review]

**Pull request merged**